xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -612,7 +612,7 @@ namespace XBMCAddon
                                                            getMusicStringArray(alt, key, value));
             else if (key == "comment")
               InfoTagMusic::setCommentRaw(musictag, value);
-            else if (key != "mediatype")
```

Comment:
This may be a possible problem. If key is `mediaType` it could cause a spurious error to be logged, right?

Reply:
The very first if is key == mediatype. This is why the warning. Key != Mediatype is redundant as it has to NOT be mediatype to get this far